### ISSUE TITLE
Allow { type: integer } for mappings when using HierarchicalRefinementFilter

### ISSUE THREAD

**Issue by voke** [searchkit/searchkit#472]
I'm using the `HierarchicalRefinementFilter` with the following mapping:
```ruby
category:{
  type:"nested",
  properties:{
    level: { type:"integer" },
    ancestors:{ type:"integer", index:"not_analyzed" },
    value:{ type:"integer", index:"not_analyzed"},
    order:{ type:"integer"}
  }
}    
```
Unlike the [example mapping](http://docs.searchkit.co/stable/components/navigation/hierarchical-refinement-filter.html) in the docs my category ids are **Integers** and I use the `translations` option to render them as text.

When I navigate down the tree and **reload the page** the categories are not selected/expanded.

I tracked it down to https://github.com/searchkit/searchkit/blob/develop/src/core/state/LevelState.ts#L30. On page reload the `this.getValue()[level]` returns an array of strings instead of integers so `indexOf()` will return false.

**Comment by ssetem** [comment]
this will be fixed in upcoming 2.3.0 release

**Comment by ssetem** [comment]
@voke 
this should be fixed in beta version 2.3.0-0

thanks!
